File: 004/PE004.c
#include <math.h>
#include <stdio.h>
#include <string.h>
#include <stdlib.h>


int main(){
	int p,ll,ul,maxnum,x,g;
	char k[20]={0}; 
	char K[20]={0};
	/* 
	P=number of digits
	ll=lover limit
	ul=upper limit
	*/
	printf("Enter the number of digits to multiply :");
	scanf("%d",&p);
	ul=powl(10,p)-1;
	ll=ul;
	x=ul;
	maxnum=0;

	while (ll!=pow(10,p-1)){
		for (int i=ul;i>ll-1;i--){
			g=i*ll;
			if (maxnum>g){
				break;
			}
			sprintf(k,"%d",g);
			int len=strlen(k);			
			int qwe=1;
			for (int j=0;j<len;j++){
					
					if (k[j]!=k[len-1-j]){
						qwe=0;
						break;
					}
			}
			//printf("%s %d %d\n",k,qwe,maxnum);
			if (qwe==1){
				if (maxnum<g){
					maxnum=g;
				}
			}
			


		}
		ll-=1;
	}
	printf("The maximum palandromic number is %d\n",maxnum );
}

File: 004/PE004.py
# Project Euler 4 --- Largest palindrome product

p=int(input("Enter the number of digits :"))
ul=(10**p) - 1
ll=ul
x=ul
maxnum=0
while ll!=10**(p-1):
    for i in range(x,ll-1,-1):
        k=str(i*ll)
        
        if k==k[::-1] and int(k)>maxnum:
            maxnum=int(k)
            
    ll-=1
print("Maximum palandromin number in rh range is : ",maxnum)
